app/validation/validator.py

```diff
@@ -506,6 +507,45 @@ def _validate_routing_rule(self, rule, answer_ids_with_group_id, block_or_group)
 
         return errors
 
+    def _validate_answer_routing_value(self, rule, answer_ids_with_parent_id):
+        errors = []
+
+        when_rule = rule.get('goto').get('when')
+        if when_rule:
+            for when_dict in when_rule:
+                amount_matched = 0
+                errors.extend(self._validate_routing_when_dict(when_dict, answer_ids_with_parent_id, amount_matched))
+
+        return errors
+
+    def _validate_routing_when_dict(self, when_dict, answer_ids_with_parent_id, amount_matched):
+        errors = []
+
+        if when_dict.get('value'):
+            rule_value = when_dict.get('value')
+            errors.extend(self._validate_routing_when_dict_value(answer_ids_with_parent_id, amount_matched, rule_value))
+
+        elif when_dict.get('values'):
+            for rule_value in when_dict.get('values'):
+                errors.extend(self._validate_routing_when_dict_value(
+                    answer_ids_with_parent_id, amount_matched, rule_value))
+
+        return errors
```

Comment:
Both these calls are doing the same thing, once for values and another for value. You can save the nesting here if you make both routes use the same call.

```python
errors = []
when_values = when_dict.get('values', [])

if not when_values:
    when_values.append(when_dict.get('value'))

for rule_value in when_values:
    valid_rule = self.is_rule_value_valid(
                    answer_ids_with_parent_id, rule_value))

   if not valid_rule:
      errors.append(f'Answer option and routing rule values mismatch, missing answer value: {rule_value}')